WordPress/gutenberg
# Background block supports: move block support defaults to gutenberg_render_background_support and revert gutenberg_get_background_support_styles

## What and how?

This PR:

1. moves block supports defaults logic (back) to `gutenberg_render_background_support` on the PHP side, and to the background hook (from style engine).
2. reverts `gutenberg_get_background_support_styles` as it's only calling the style engine, and moving the style engine call to the theme json class.

The consequences are:

1. Block supports receive specific default value logic only (setting "cover" for "background-size" etc)

Context:

- https://github.com/WordPress/gutenberg/pull/59454#issuecomment-1975790708

> [!IMPORTANT]
> Changes in this PR only affect the frontend. Harmonizing the site editor with these changes is happening over in https://github.com/WordPress/gutenberg/pull/60008


## Why?

To isolate specific property mutations to block supports.

When adding backgrounds to blocks in the editor, e.g., Group block, having defaults allows users to see the scaled image immediately as the containing element might be too small or large for the source image.

For site backgrounds in global styles, these defaults may not make sense, especially if theme devs deliberately leave them out in theme.json.

As for reverting `gutenberg_get_background_support_styles`, it's code that's not longer required. It hasn't been in Gutenberg for long and is not in WordPress Core. Let's re-add it as we need it, which might be soon.

## Testing Instructions

Test the changes on a published post in the frontend.

1. When adding a background image to theme JSON (example theme.json below), the default background size should not be set.
3. For Group blocks, the default value should be "cover" UNLESS the `supports.background.backgroundSize` is set to another defaults, e.g., `"contain"`, in which case it'll be "contain"

Example theme.json

```json
{
	"$schema": "../../schemas/json/theme.json",
	"version": 2,
	"settings": {
		"appearanceTools": true,
		"background": {
			"backgroundImage": true,
			"backgroundSize": true
		}
	},
	"styles": {
		"color": {
			"background": "tomato"
		},
		"background": {
			"backgroundImage": "url('https://i0.wp.com/wordpress.org/files/2023/12/sotw-dotorg-drawer.png?w=1807&ssl=1')"
		}
	}
}

```

## Screenshots or screencast 




https://github.com/WordPress/gutenberg/assets/6458278/0acd750a-5f62-486a-bdcd-81c01de3613b
